### ISSUE TITLE
autocomplete is blocked (happens in demo app to)

### ISSUE THREAD

**Issue by TomJannes** [marmelab/react-admin#2346]
**What you were expecting:**
Being able to type in the autocompleteinput

**What happened instead:**
autocompleteinput is blocked, i can not change the value. this also happens in the demo application.

**Steps to reproduce:**
```
<ReferenceInput label="Company" source="companyId" reference="companies" filterToQuery={searchText => ({ name: searchText })}>
    <AutocompleteInput optionText="name" optionValue="id"/>
</ReferenceInput>
```

**Environment**

* React-admin version: 2.3.1
* React version: 16.5.2
* Browser: chrome
* Stack trace (in case of a JS error): no error

**Comment by AkselsLedins** [comment]
2.3.2 aswell

We are experiencing this since a week approx. (even in production :crying_cat_face: )

Demo (try to select a customer):
https://marmelab.com/react-admin-demo/#/commands

![demo-bug](https://user-images.githubusercontent.com/18367406/45886073-67ecbb80-bdb8-11e8-81f9-5153fb402d0f.gif)

@valours @ArnaudD

**Comment by AkselsLedins** [comment]
@kamil-borecki Thank you it helped

**Comment by TomJannes** [comment]
Solved my problem, thanks!

**Comment by afilp** [comment]
@fzaninotto @djhi I am also in need to change this behavior because I cannot type anything else if one item is already matched. So, I will have to Find/Replace all `<AutocompleteInput>` to add this custom function. 

It is easy to do, but since this will be a common request by others (I believe so), can we have a different behavior there? Like `inputValueMatcher = {false}`.

I would prefer to be the opposite and have this as a default but I understand that this would break previous implementations. If the `false` was the default instead, then the user would have to explicitly specify the `inputValueMatcher` either as `true` (i.e. using the default function that you already have) OR `() => {\\do different things here}`.

**Comment by jheiska** [comment]
Still have this problem in 2.4.4 Is anyone working on this? It's a very annoying bug. Some other problems with the AutocompleteInput component, too. I'll list a few here. The problems are present when I have the input inside a ReferenceInput. I only use autocomplete inside reference, so not sure where the problem lies.

- When you have selected a value, basically the only way to get back to the "autocomplete mode" is to select the whole text and delete it - you cannot use backspace to remove some letters from the end. I think this is what the original issue referred to. 

- Sometimes when you are writing on the input, it randomly selects a value. 

- Sometimes the selections pop up on top of the input element and you can't access it and sometimes completely out of the screen (it was a while ago when this happened, not sure if it's a problem in the current version) - this is dependent on the screen size.

- Also noticed just now, playing around with the input, it doesn't always work correctly - I get a list of names that don't include the text I wrote in the input at all. Actually this seems to be quite a constant problem. I was using the customer filter in https://marmelab.com/react-admin-demo/#/commands for this testing.

**Comment by fzaninotto** [comment]
@jheiska if you're in a hurry, I suggest you try and fix the bug in the react-admin code and submit a pull request. That's the beauty of open-source: you're never blocked by the editor / maintainers.

**Comment by jheiska** [comment]
@fzaninotto if I have the time, I will see if I can locate and/or fix the problem(s).

**Comment by fzaninotto** [comment]
Fixed by #2833

**Comment by e-ndrus** [comment]
I don't think it is fixed by #2833. My version of react-admin is "react-admin": "^2.6.1". I applied the fix, but the problem persists with the following setup:
` <ReferenceInput sort={{ field: 'product_en', order: 'ASC' }} label="description" source="product_en" reference="products" allowEmpty>
                <AutocompleteInput optionValue="_id" inputValueMatcher={() => null} optionText="product_en" />
              </ReferenceInput>`
When I select a value with autocompleteInput, I can't select another one even if I remove/delete the selection. The only way how to select another value is to refresh the page.
